Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp

```diff
@@ -4043,7 +4043,6 @@ void BlockNode::emitBytecode(BytecodeGenerator& generator, RegisterID* dst)
 
 void EmptyStatementNode::emitBytecode(BytecodeGenerator&, RegisterID*)
 {
-    RELEASE_ASSERT(needsDebugHook());
```

Comment:
Scratch that, why was this there in the first place?

Reply:
Before this change, `EmptyStatementNode` was only generated for `;` and its location was always recorded as pause point for debugger purposes as it can be in a loop, which is what this assert is testing. Basically it tests `shouldSetPauseLocation = true;` in `parseStatement()`.

This patch adds another `EmptyStatementNode` usage, making the not-very-useful assert now incorrect.